clustersConfig.py

```diff
@@ -42,7 +42,7 @@ class NodeConfig:
     preallocated: str
     os_variant: str
     ip: Optional[str] = None
-    bmc_ip: Optional[str] = None
+    bmc_ip: str = ""
```

Comment:
We need to remove the default value here or else the user has the option not to provide it

Reply:
https://github.com/bn222/cluster-deployment-automation/blob/main/clustersConfig.py#L171, here we are unpacking the dictionary w to keyword arguments using the "**" operation which are passed to the NodeConfig constructor. We need to make sure bmc_ip is a required argument here.

If you are curious how this works: https://docs.python.org/3/library/dataclasses.html.

You will also need to put required arguments before optional or you will get a TypeError.

Reply:
Aahh disregard, just tested this and this is not completely accurate

Reply:
We should however still remove the default value here.

Reply:
We may want to add manual checks within the init function to ensure the required arguments are provided. If we don't we will not get an error until they are accessed later.